### PR TITLE
Moved down_count to CoreTiming

### PR DESCRIPTION
**Pull request by huwpascoe**
It's needed for testing.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/2973)

### CODE REVIEW

File: src/tests/core/arm/dyncom/arm_dyncom_vfp_tests.cpp [hunk 1]
```diff
@@ -29,7 +30,7 @@ TEST_CASE("ARM_DynCom (vfp): vadd", "[arm_dyncom]") {
     }};
 
     for (const auto& test_case : test_cases) {
-        dyncom.down_count = 1000; // Ensure that CoreTimeing will not be called.
+        CoreTiming::SetDownCount(1000); // Ensure that CoreTimeing will not be called.
```
Comment (Subv) [inline src/tests/core/arm/dyncom/arm_dyncom_vfp_tests.cpp]: Remove this call, it's no longer needed

File: src/core/core_timing.cpp [hunk 2]
```diff
@@ -185,8 +188,19 @@ void Shutdown() {
     }
 }
 
+void SetDownCount(u64 value) {
```
Comment (Subv) [inline src/core/core_timing.cpp]: This function is not needed

File: src/tests/core/arm/dyncom/arm_dyncom_vfp_tests.cpp (hunk 1 above)
Comment (huwpascoe) [reply src/tests/core/arm/dyncom/arm_dyncom_vfp_tests.cpp]: Good

**Comment by huwpascoe** [comment]
amended

**Comment by tywald** [comment]
What's been said on Discord should probably be copy-pasted in the description so the information is tied here.

**Comment by huwpascoe** [comment]
Um, all that was said was it's needed for testing. Should I add that?

**Comment by tywald** [comment]
Better than "as mentioned in discord" if you will look back at this PR one year from now.

**Comment by MerryMage** [comment]
Note for posterity: Decouples CoreTiming from CPU emulation, making it easier for unit tests to be written.

**Pull request merged**